config/configuration_file.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2019 Sonatype Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package config
+
+import (
+	"github.com/sirupsen/logrus"
+	"github.com/sonatype-nexus-community/go-sona-types/ossindex/types"
+	"gopkg.in/yaml.v3"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+type ChequeConfig struct {
+	OSSIndexConfig OSSIConfig
+	IQConfig IQConfig
+}
+
+type OSSIConfig struct {
+	Username string `yaml:Username`
+	Password string `yaml:Password`
+}
+
+type IQConfig struct {
+	Server   string `yaml:"Server"`
+	Username string `yaml:"Username"`
+	Token    string `yaml:"Token"`
+}
+
+type Config struct {
+	logger *logrus.Logger
+}
+
+func New(logger *logrus.Logger) *Config {
+	return &Config{logger: logger}
+}
+
+func (c Config) CreateOrReadConfigFile() ChequeConfig {
+
+	createDirectory(c.logger, types.IQServerDirName)
+	createDirectory(c.logger, types.OssIndexDirName)
+
+	if !fileExists(getIQConfig()) {
+		writeDefaultIQConfig(c.logger)
+	}
+	if !fileExists(getOssiConfig()) {
+		writeDefaultOssiConfig(c.logger)
+	}
+
+	return readConfig(c.logger)
+}
+
+func createDirectory(logger *logrus.Logger, directory string) {
+	home, _ := os.UserHomeDir()
+	fileDirectory := filepath.Join(home, directory)
+	err := os.MkdirAll(fileDirectory, os.ModePerm)
+	if err != nil {
+		logger.Error("Couldn't create config directory: " + fileDirectory)
+	}
+}
+
+//Gets the default location for the config file
+func getIQConfig() string {
+	home, _ := os.UserHomeDir()
+	filePath := filepath.Join(home,types.IQServerDirName, types.IQServerConfigFileName)
+	return filePath
+}
+
+func getOssiConfig() string {
+	home, _ := os.UserHomeDir()
+	filePath := filepath.Join(home,types.OssIndexDirName, types.OssIndexConfigFileName)
```

Comment:
Since `filepath.Join` does not return an error you can just do `return filepath.Join(home,types.OssIndexDirName, types.OssIndexConfigFileName)`

Alternatively you could do:

```
func getIQConfig() (filepath string) {
	home, _ := os.UserHomeDir()
	filePath = filepath.Join(home,types.IQServerDirName, types.IQServerConfigFileName)
	return
}
```

^ that's what we call a naked return and a named return value (I love them because you can skip variable initialization)